indico/modules/events/papers/controllers/display.py

```diff
@@ -53,3 +61,114 @@ def _process(self):
     def _process(self):
         return NotImplementedError
+
+
+class RHDisplayJudgingArea(RHPapersBase):
+    def _checkProtection(self):
+        if not session.user:
+            raise Forbidden
+            RHPapersBase._checkProtection(self)
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        self.list_generator = PaperJudgingAreaListGeneratorDisplay(event=self.event_new, user=session.user)
+
+    def _process(self):
+        return WPDisplayJudgingArea.render_template('display/judging_area.html', self._conf, event=self.event_new,
+                                                    **self.list_generator.get_list_kwargs())
+
+
+class RHDisplayCustomizeJudgingAreaList(RHPapersBase):
+    """Display dialog with filters"""
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        self.list_generator = PaperJudgingAreaListGeneratorDisplay(event=self.event_new, user=session.user)
+
+    def _process_GET(self):
+        list_config = self.list_generator.list_config
+        return WPDisplayJudgingArea.render_template('management/assignment_list_filter.html', self._conf,
+                                                    event=self.event_new,
+                                                    static_items=self.list_generator.static_items,
+                                                    filters=list_config['filters'],
+                                                    visible_items=list_config['items'])
+
+    def _process_POST(self):
+        self.list_generator.store_configuration()
+        return jsonify_data(flash=False, **self.list_generator.render_list())
+
+
+class RHPapersActionsBase(RHPapersBase):
+    """Base class for RHs performing actions on selected contributions in judging area (display)"""
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        ids = map(int, request.form.getlist('contribution_id'))
+        self.contributions = Contribution.query.with_parent(self.event_new).filter(Contribution.id.in_(ids)).all()
+        self.role = request.args.get('role')
+
+
+class RHDisplayBulkPaperJudgment(PaperJudgmentMixin, RHPapersActionsBase):
+    def _checkParams(self, params):
+        RHPapersActionsBase._checkParams(self, params)
+        self.list_generator = PaperJudgingAreaListGeneratorDisplay(event=self.event_new, user=session.user)
+
+
+class RHDisplayPapersDownload(PapersDownloadAttachmentsMixin, RHPapersActionsBase):
+    pass
+
+
+class RHJudgingAreaAssignDialog(RHPapersBase):
+    """Dialog with person list to assign/unassign paper reviewing roles"""
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        ids = map(int, request.form.getlist('contribution_id'))
+        self.contributions = Contribution.query.with_parent(self.event_new).filter(Contribution.id.in_(ids)).all()
+        self.role = request.args.get('role')
```

Comment:
`self.role = PaperReviewingRole[request.args['role']]` - then you have the enum item itself instaed of just its string name